src/components/JuiceModal.tsx

```diff
@@ -0,0 +1,19 @@
+import { Modal, ModalProps } from 'antd'
+import { ThemeContext } from 'contexts/themeContext'
+import { useContext } from 'react'
+import { classNames } from 'utils/classNames'
+
+/**
+ * A wrapper around the Ant Design Modal component that adds the correct dark
+ * mode class for tailwind.
+ *
+ */
+export const JuiceModal = (props: ModalProps) => {
+  const { isDarkMode } = useContext(ThemeContext)
+  return (
+    <Modal
+      {...props}
+      className={classNames(props.className, isDarkMode ? 'dark' : undefined)}
```

Comment:
**question** curious why we need this if the root-level `dark` class is already set.

Reply:
These are set independent of the root. modal/tooltips/etc float above the root in a sibling div

Reply:
ah good one, of course. nice

Reply:
begs the question though: can we set `dark` on the `<body>` elem in Next? 🤔

Reply:
Maybe this : https://nextjs.org/docs/advanced-features/custom-document

Reply:
Hmm dont think that'll work cos its serverside only.

WDYT about something like this instead

```js
  useEffect(() => {
    document.body.className = isDarkMode ? 'dark' : '';
  });
```